src/main/java/Duke.java

```diff
@@ -1,10 +1,247 @@
+        }
+
+        String[] tmp = s[1].split("/at");
+        if (tmp.length < 2) {
+            throw new DukeException("Event description is missing the /at tag.");
+        }
+
+        try {
+            Event task = new Event(tmp[0].strip(), tmp[1].strip());
+            db.add(task);
+
+            System.out.println("Got it. I added this event:");
+            System.out.println("\t" + task);
+            System.out.printf("Now you have %d tasks in the list.%n", db.size());
+        } catch (DateTimeParseException err) {
+            throw new DukeException("Invalid datetime argument passed to event.\nPlease enter in YYYY-MM-DD HHMM format.");
+        }
+    }
+
+    /**
+     * Adds a new Deadline task.
+     * @param s Input task with description
+     */
+    private void addDeadline(String[] s) throws DukeException {
+        if (s.length < 2 || s[1].strip().equals("")) {
+            throw new DukeException("Deadline description is empty.");
+        }
+
+        String[] tmp = s[1].split("/by");
+        if (tmp.length < 2) {
+            throw new DukeException("Deadline description is missing the /by tag.");
+        }
+
+        try {
+            Deadline task = new Deadline(tmp[0].strip(), tmp[1].strip());
+            db.add(task);
+
+            System.out.println("Got it. I added this deadline:");
+            System.out.println("\t" + task);
+            System.out.printf("Now you have %d tasks in the list.%n", db.size());
+        } catch (DateTimeParseException err) {
+            throw new DukeException("Invalid datetime argument passed to deadline.\nPlease enter in YYYY-MM-DD HHMM format.");
+        }
+    }
+
+    private void delete(String[] userInput) {
+        Task tmp = db.remove(Integer.parseInt(userInput[1]) - 1);
+        System.out.println("OK, I've removed this task:");
+        System.out.println("\t" + tmp);
+        System.out.printf("Now you have %d tasks in the list.", db.size());
+    }
+
+    /**
+     * Stores the current list of Tasks to data/duke.txt.
+     */
+    private void write() throws IOException {
+        FileWriter fw = new FileWriter("data/duke.txt");
+        for (Task t: db) {
+            fw.write(t.toStringWritable().strip());
+            fw.write("\n");
+        }
+        fw.close();
+    }
+
+    private void load() throws IOException {
+        Scanner s = new Scanner(new File("data/duke.txt"));
+        while (s.hasNext()) {
+            String[] tmp = s.nextLine().split("\\|");
+            for (int i = 0; i < tmp.length; i++) {
+                tmp[i] = tmp[i].strip();
+            }
+
+            switch (tmp[0]) {
+                case "T":
+                    db.add(new Todo(tmp));
+                    break;
+                case "D":
+                    db.add(new Deadline(tmp));
+                    break;
+                case "E":
+                    db.add(new Event(tmp));
+                    break;
```

Comment:
I like how you used a switch statement here! But take note that we are not supposed to have indentations for case clauses!